File: TheSweeper/src/Utils/JwtGenerator.py
import jwt
import datetime
import pytz
from settings.AppSettings import SECRET_KEY, TIMEZONE, JWT_TTL

class JwtGenerator:
    def __init__(self, id: int):
        self.id = id
        self.time_zone = pytz.timezone(TIMEZONE)

    def jwt(self):
        now = datetime.datetime.now(self.time_zone)
        payload = {
            "sub": self.id,
            "iat": now,  
            "exp": now + datetime.timedelta(minutes=int(JWT_TTL)),  
            "admin": True,  
        }
        return jwt.encode(payload=payload, key=SECRET_KEY, algorithm="HS256")

File: app/utils/JwtGenerator.php
<?php

namespace App\utils;

use Carbon\Carbon;
use \Firebase\JWT\JWT;

class JwtGenerator{
    
    const SECRET_KEY = 'JWT_SECRET';  
    const TTL = 'JWT_TTL'; 

    private $id;
    private $now;
    private $expiration;

    public function __construct($id)
    {
        $this->id = $id;
        $this->now = Carbon::now()->timestamp;
        $this->expiration = Carbon::now()->addMinutes((int)env(self::TTL))->timestamp; 
    }

    public function jwt()
    {
        
        $payload = [
            "sub" => $this->id,  
            "iat" => $this->now , 
            "exp" => $this->expiration 
        ];

        
        return JWT::encode($payload, env(self::SECRET_KEY), 'HS256');
    }
}
